osf/models/node.py

```diff
@@ -438,6 +438,11 @@ def is_preprint_orphan(self):
         return False
 
     @property
+    def has_published_preprint(self):
+        PreprintService = apps.get_model('osf.PreprintService')
+        return PreprintService.objects.filter(node_id=self.id, is_published=True).exists()
```

Comment:
Use the `related_name` (aka backref or reverse accessor) rather than querying on the related model.

```python
class PreprintService(GuidMixin, BaseModel):
    node = models.ForeignKey('osf.AbstractNode',
                             related_name='preprints',...)


# Yes
self.preprints.filter(...)

# No
PreprintService.objects.filter(node=self, ...)
```